src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertyAndValueDeserializer.cs

```diff
@@ -491,7 +494,21 @@ protected static void ValidateExpandedNestedResourceInfoPropertyValue(IJsonReade
                 // Expanded resource (null or non-null)
                 if (isCollection == true)
                 {
-                    throw new ODataException(ODataErrorStrings.ODataJsonLightResourceDeserializer_CannotReadCollectionNestedResource(nodeType, propertyName));
+                    if (typeReference.IsODataPrimitiveTypeKind() ||
+                        typeReference.IsODataTypeDefinitionTypeKind() ||
+                        typeReference.IsODataEnumTypeKind() ||
+                        typeReference.IsODataComplexTypeKind() ||
+                        typeReference.IsUntyped() ||
+                        typeReference.IsNonEntityCollectionType())
+                    {
+                        ReaderValidationUtils.ValidateNullValue(typeReference, true,

```

Comment:
Any reason to use `ValidateNullValue` directly via `ReaderValidationUtils` instead of using it via `ReaderValidator` which binds to `ODataMessageReaderSettings`?
`ReaderValidator` uses settings from `ODataMessageReaderSettings` to determine the value for `enablePrimitiveTypeConversion`, which in this case you are ignoring and forcing to always be true

Reply:
Have used the <code> ValidateNullValue </code> directly via <code> ReaderValidationUtils </code> because at this time I know am validating a collection of a complex type and not of a primitive type. It really does not matter the value that I pass for the 
<code> enablePrimitiveTypeConversion </code> at this point. It can as well be false. it won't have any effect on the method.